Modules/ITS/src/ITSChipStatusTask.cxx

```diff
@@ -0,0 +1,178 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   ITSChipStatusTask.cxx
+/// \author My Name
+///
+
+#include <TCanvas.h>
+#include <TH1.h>
+
+#include "QualityControl/QcInfoLogger.h"
+#include "ITS/ITSChipStatusTask.h"
+#include <Framework/InputRecordWalker.h>
+#include <Framework/DataRefUtils.h>
+#include "TLine.h"
+#include "TLatex.h"
+
+namespace o2::quality_control_modules::its
+{
+ITSChipStatusTask::ITSChipStatusTask()
+  : TaskInterface()
+{
+}
+
+ITSChipStatusTask::~ITSChipStatusTask()
+{
+  for (int i = 0; i < 3; i++) {
+    delete ChipsStack[i];
+    delete DeadChips[i];
+  }
+}
+
+void ITSChipStatusTask::initialize(o2::framework::InitContext& /*ctx*/)
+{
+  ILOG(Debug, Devel) << "initializing the ITSChipStatusTask" << ENDM;
+  getParameters();
+
+  int Layer_Draw = 0;
+  for (int i = 0; i < 3; i++) {
+
+    ChipsStack[i] = new Stack(nQCCycleToMonitor, ChipsBoundaryBarrels[i + 1] - ChipsBoundaryBarrels[i]);
+
+    DeadChips[i] = new TH2D(Form("DeadChips%s", BarrelNames[i].Data()), Form("Dead Chips in %s", BarrelNames[i].Data()), nQCCycleToMonitor, 0, nQCCycleToMonitor, ChipsBoundaryBarrels[i + 1] - ChipsBoundaryBarrels[i], 0, ChipsBoundaryBarrels[i + 1] - ChipsBoundaryBarrels[i]);
+    setAxisTitle(DeadChips[i], "QC Cycle ID", "");
+
+    int iLayerBegin, iLayerEnd;
+    if (i == 0) {
+      iLayerBegin = 0;
+      iLayerEnd = 2;
+    } else if (i == 1) {
+      iLayerBegin = 3;
+      iLayerEnd = 4;
+    } else {
+      iLayerBegin = 5;
+      iLayerEnd = 6;
+    }
+    const int ChipBoundary[8] = { 0, 108, 252, 432, 3120 - 432, 6480 - 432, 14712 - 6480, 24120 - 6480 }; // needed for drawing labels and lines for histogram
+    const int StaveBoundary[NLayer] = { 0, 12, 28, 0, 24, 0, 42 };
```

Comment:
`ChipBoundary[8]` and `StaveBoundary[NLayer] are already in the header file. I would move these two to the header file if you really needed them (eventually renaming them properly).